lib/internal/Magento/Framework/App/Response/Http.php

```diff
@@ -131,7 +133,7 @@ public function sendVary()
      */
     public function setPublicHeaders($ttl)
     {
-        if ($ttl < 0 || !preg_match('/^[0-9]+$/', $ttl)) {
+        if ($ttl < 0 || !preg_match('/^[0-9]+$/', (string) $ttl)) {
             throw new \InvalidArgumentException('Time to live is a mandatory parameter for set public headers');
         }
         $this->setHeader('pragma', 'cache', true);
```

Comment:
Added compulsive type here for input `$ttl` 
CC: @eduard13

Reply:
But, have you tried to update the Unit Test that I mentioned, to see if that's the reason of why the test was previously failing (referencing the comment above)?

Reply:
Test file still remain i don't change the null test case! Only change in origin file. Re-run unit test passed in my local